spec/workers/concerns/code_analysis_mixin_spec.rb

```diff
@@ -8,34 +8,188 @@
   end
   subject { test_class.new }
 
-  context "#merged_linter_results" do
+  describe "#merged_linter_results" do
     it "should always return a hash with a 'files' and 'summary' key, even with no cops running" do
-      expect(Linter::Rubocop).to receive(:new).and_return(double(:run => nil))
-      expect(Linter::Haml).to receive(:new).and_return(double(:run => nil))
-      expect(Linter::Yaml).to receive(:new).and_return(double(:run => nil))
-
-      expect(subject.merged_linter_results).to eq("files"=>[], "summary"=>{"inspected_file_count"=>0, "offense_count"=>0, "target_file_count"=>0})
+      allow(subject).to receive(:pronto_result).and_return([])
+      expect(subject.merged_linter_results).to eq("files" => [], "summary" => {"inspected_file_count" => 0, "offense_count" => 0, "target_file_count" => 0})
     end
+  end
 
-    it "merges together with one result" do
-      expect(Linter::Rubocop).to receive(:new).and_return(double(:run => example_rubocop_result))
-      expect(Linter::Haml).to receive(:new).and_return(double(:run => nil))
-      expect(Linter::Yaml).to receive(:new).and_return(double(:run => nil))
+  describe "#run_all_linters" do
+    let(:item_rubocop) { dup }
```

Comment:
I've never seen `dup` like this before...what does this do?

Reply:
@Fryguy it's a dummy object for mocking

Reply:
Is it the same as just calling `double`?

Reply:
I don't think this is doing what you think it does  (I can't find any docs on RSpec providing a dup method).  If anything it's probably calling Object#dup on the current context?  I think I would prefer `double` here, so that it's explicit that we want a mock object.